examples/ppo/ppo.py

```diff
@@ -0,0 +1,117 @@
+import argparse
+from importlib.resources import path
+import pathlib
+import torch
+
+import hivemind
+from hivemind import Float16Compression, SizeAdaptiveCompression, Uniform8BitQuantization
+
+from stable_baselines3 import PPO
+from stable_baselines3.common.env_util import make_atari_env
+from stable_baselines3.common.vec_env import VecFrameStack
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--n-steps', type=int, default=128, help='Number of rollout steps per each agent')
+    parser.add_argument('--n-envs', type=int, default=8, help='Number of training envs')
+    parser.add_argument('--batch-size', type=int, default=256)
+    parser.add_argument('--target-batch-size', type=int, default=32768)
+    parser.add_argument('--n-epochs', type=int, default=1, help='Number of training epochs per each rollout')
+    parser.add_argument('--learning-rate', type=float, default=2.5e-4)
+    parser.add_argument('--tb-logs-path', type=pathlib.Path, default='./logs', help='Path to tensorboard logs folder')
+    parser.add_argument('--experiment-prefix', type=str, help='Experiment prefix for tensorboard logs')
+    parser.add_argument('--initial-peers', nargs='+', default=[])
+    parser.add_argument('--averaging-compression', action='store_true')
+    args = parser.parse_args()
+    return args
+
+def generate_experiment_name(args):
+    exp_name_dict = {
+        'bs': args.batch_size,
+        'target_bs': args.target_batch_size,
+        'n_envs': args.n_envs,
+        'n_steps': args.n_steps,
+        'n_epochs': args.n_epochs,
+    }
+
+    exp_name = [f'{key}-{value}' for key, value in exp_name_dict.items()]
+    exp_name = '.'.join(exp_name)
+
+    if args.experiment_prefix:
+        exp_name = f'{args.experiment_prefix}.{exp_name}'
+    exp_name = exp_name.replace('000.', 'k.')
+    return exp_name
+
+
+class AdamWithClipping(torch.optim.Adam):
```

Comment:
note to @mryab : we've recently merged the same clipping functionality here:
https://github.com/learning-at-home/hivemind/blob/master/hivemind/moe/server/layers/optim.py#L48

Would you prefer if we...
- keep everything as is, accept some code duplication?
- extract moe.server.layers.optim to utils.optim and use it here?
- keep wrapper in hivemind.optim and import from there?
- insert your option here :)

Reply:
I'm 50:50 between the "keep here, accept duplication" and "move OptimizerWrapper and ClippingWrapper to hivemind.optim.wrappers" solutions, so ultimately, it's @foksly's call